src/database/database-mariadb.cpp

```diff
@@ -0,0 +1,1407 @@
+
+	try {
+
+		stmtLoadBlock->setInt(1, htonl(pos.X));
+		stmtLoadBlock->setInt(2, htonl(pos.Y));
+		stmtLoadBlock->setInt(3, htonl(pos.Z));
+		
+		std::unique_ptr<sql::ResultSet> resBlock(stmtLoadBlock->executeQuery());
+
+		if (resBlock->next()) {
+
+			std::stringstream in;
+			in << resBlock->getBlob("data")->rdbuf();
+			std::string data = std::move(in).str();
+			*block = data;
+
+		} else {
+
+			block->clear();
+
+		}
+
+		resBlock->close();
+		stmtLoadBlock->clearParameters();
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+bool MapDatabaseMariaDB::saveBlock(const v3s16 &pos, const std::string &data) {
+
+	checkConnection();
+
+	try {
+
+		size_t max_size = 4294967295;
+		if (data.size() > max_size) {
+
+			errorstream << "[MariaDB] Error: refusing to save block because data would be truncated: " << data.size() << " > " << max_size << std::endl;
+			return false;
+
+		}
+
+		std::istringstream _data(data);
+
+		stmtSaveBlock->setInt(1, htonl(pos.X));
+		stmtSaveBlock->setInt(2, htonl(pos.Y));
+		stmtSaveBlock->setInt(3, htonl(pos.Z));
+		stmtSaveBlock->setBlob(4, &_data);
+		stmtSaveBlock->executeUpdate();
+		stmtSaveBlock->clearParameters();
+
+		std::int32_t rows_affected = stmtSaveBlock->getUpdateCount();
+
+		return (rows_affected > 0);
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+/*
+ *
+ * PlayerDatabaseMariaDB -- The MariaDB Player Database Backend
+ * 
+ */
+
+
+PlayerDatabaseMariaDB::PlayerDatabaseMariaDB(const std::string &connect_string)
+: Database_MariaDB(connect_string, "_player"), PlayerDatabase() {
+	
+	// connect to database
+	connect();
+
+	// create tables if they do n
+	createTable("player",
+		"name VARCHAR(60),"
+		"pitch DECIMAL(15, 7) NOT NULL,"
+		"yaw DECIMAL(15, 7) NOT NULL,"
+		"posX DECIMAL(15, 7) NOT NULL,"
+		"posY DECIMAL(15, 7) NOT NULL,"
+		"posZ DECIMAL(15, 7) NOT NULL,"
+		"hp INT NOT NULL,"
+		"breath INT NOT NULL,"
+		"creation_date DATETIME NOT NULL DEFAULT NOW(),"
+		"modification_date DATETIME NOT NULL DEFAULT NOW(),"
+		"PRIMARY KEY (name)"
+	);
+
+	createTable("player_inventories",
+		"player VARCHAR(60),"
+		"inv_id INT NOT NULL,"
+		"inv_width INT NOT NULL,"
+		"inv_name TEXT NOT NULL DEFAULT '',"
+		"inv_size INT NOT NULL,"
+		"PRIMARY KEY(player, inv_id),"
+		"CONSTRAINT player_inventories_fkey FOREIGN KEY (player) REFERENCES player (name) ON DELETE CASCADE"
+	);
+
+	createTable("player_inventory_items",
+		"player VARCHAR(60),"
+		"inv_id INT NOT NULL,"
+		"slot_id INT NOT NULL,"
+		"item TEXT NOT NULL DEFAULT '',"
+		"PRIMARY KEY(player, inv_id, slot_id),"
+		"CONSTRAINT player_inventory_items_fkey FOREIGN KEY (player) REFERENCES player (name) ON DELETE CASCADE"
+	);
+
+	createTable("player_metadata",
+		"player VARCHAR(60),"
+		"attr VARCHAR(256) NOT NULL,"
```

Comment:
put TEXT for attr too

Reply:
Fixed in latest commit. I made attr a VARCHAR because primary key is (player, attr) in Postgre implementation, however upon review, I see that this was a mistake because all player_metadata queries only use name, so Postgre's index is screwed up.